core/domain/src/commonMain/kotlin/com/mifos/core/domain/useCases/UploadClientImageUseCase.kt

```diff
@@ -26,13 +27,10 @@ class UploadClientImageUseCase(
     private val repository: ClientDetailsRepository,
 ) {
 
-    operator fun invoke(id: Int, image: MultiPartFormDataContent): Flow<DataState<String>> = flow {
-        try {
-            emit(DataState.Loading)
+    operator fun invoke(id: Int, image: MultiPartFormDataContent): Flow<DataState<String>> {
+        return flow {
             repository.uploadClientImage(id, image)
-            DataState.Success(getString(Res.string.core_domain_client_image_uploaded_successfully))
-        } catch (e: Exception) {
-            DataState.Error(e)
-        }
+            emit(getString(Res.string.core_domain_client_image_uploaded_successfully))
+        }.asDataStateFlow()
```

Comment:
@revanthkumarJ `asDataStateFlow` doesn't work this way. Right now, `repository.uploadClientImage(...)` could fail silently, but Success is always emitted.
our `asDataStateFLow` takes a flowable and return three state: Success, Loading or Error.

so the previous logic was right
```
operator fun invoke(id: Int, image: MultiPartFormDataContent): Flow<DataState<String>> = flow {
    emit(DataState.Loading) // Always emitted first

    try {
        repository.uploadClientImage(id, image)
        emit(DataState.Success(getString(Res.string.core_domain_client_image_uploaded_successfully)))
    } catch (e: Exception) {
        emit(DataState.Error(e))
    }
}
```